pandas/core/indexes/multi.py

```diff
@@ -1308,7 +1308,10 @@ def _get_grouper_for_level(self, mapper, level):
             # Remove unobserved levels from level_index
             level_index = level_index.take(uniques)
 
-        grouper = level_index.take(codes)
+        if len(level_index):
+            grouper = level_index.take(codes)
+        else:
+            grouper = level_index.take(codes, fill_value=True)
```

Comment:
hmm, can you eliminate the branch, and just always pass ``fill_value=True`` ?

Reply:
If remove the branch, for example,
```
df = DataFrame([["x", 1, 10], ["y", 2, 20]], columns=["A", "B", "C"]).set_index(["A", "B"])
result = df.groupby(level=["A", "B"]).sum() 
```
raise exception.

Reply:
actually i think this is incorrect 

@WillAyd wasn’t done here

this branch can be simplified i think

Reply:
Sorry misread the approval above

Reply:
@proost do you mind simplifying this in a follow up PR?

Reply:
@jreback 

I don't understand how to simplify. give me the hint.